tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -351,6 +351,17 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
             TraceId128BitLoggingEnabled = config
                                          .WithKeys(ConfigurationKeys.FeatureFlags.TraceId128BitLoggingEnabled)
                                          .AsBool(false);
+
+            // we "enrich" with these values which aren't _strictly_ configuration, but which we want to track as we tracked them in v1
+            telemetry.Record(ConfigTelemetryData.NativeTracerVersion, Instrumentation.GetNativeTracerVersion(), recordValue: true, ConfigurationOrigins.Default);
+            telemetry.Record(ConfigTelemetryData.FullTrustAppDomain, value: AppDomain.CurrentDomain.IsFullyTrusted, ConfigurationOrigins.Default);
+
+            if (AzureAppServiceMetadata is not null)
+            {
+                telemetry.Record(ConfigTelemetryData.AasConfigurationError, AzureAppServiceMetadata.IsUnsafeToTrace, ConfigurationOrigins.Default);
```

Comment:
I don't think it makes sense as a metric, because it's only going to be checked once at app startup 🤔 What sort of metric would it be? Config makes sense to me here tbh

Reply:
I believe error scenarios (including setup one) could be monitored through metrics. We have a metric for "exceptions" maybe we can use it. Or if we want "exceptions" to be only about exceptions, we could have a generic metric about "error" with a errortype tag? 🤷 
I think metrics will give us better error observability. But one doesn't exclude the other.
And anyway, it is out of scope of this PR as you're implementing V2 with iso functionnality. So Resolving the comment.